doc/user-guide/How-to-build.md

```diff
@@ -34,6 +34,13 @@ $ xcode-select --install # install compilation tools
 $ brew install git erlang openssl
 ```
 
+If you are planning to use ODBC to connect to MSSQL:
```

Comment:
This is not `if`. Below libs are need to compile current MongooseIM on MacOS X even if user is not going to use ODBC and MSSQL.

Reply:
Correction, `unixodbc` is needed to compile, `freetds` is needed only to connect to MSSQL

Reply:
But even though, it doesn't work on MacOS X High Sierra :( that's really sad we make our own life harder... Not everyone is using El Capitan or Sierra :(

Reply:
```===> /Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:344:14: warning: implicit declaration of function 'connect_to_erlang' is invalid in C99 [-Wimplicit-function-declaration]
    socket = connect_to_erlang(port);
             ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:344:12: error: non-object type 'int (int, int, int)' is not assignable
    socket = connect_to_erlang(port);
    ~~~~~~ ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:345:11: warning: implicit declaration of function 'receive_msg' is invalid in C99 [-Wimplicit-function-declaration]
    msg = receive_msg(socket);
          ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:345:9: warning: incompatible integer to pointer conversion assigning to 'byte *' (aka 'char *') from 'int' [-Wint-conversion]
    msg = receive_msg(socket);
        ^ ~~~~~~~~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:354:5: warning: implicit declaration of function 'close_socket' is invalid in C99 [-Wimplicit-function-declaration]
    close_socket(socket);
    ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:378:12: error: non-object type 'int (int, int, int)' is not assignable
    socket = connect_to_erlang(port);
    ~~~~~~ ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:381:17: warning: incompatible integer to pointer conversion assigning to 'byte *' (aka 'char *') from 'int' [-Wint-conversion]
        request_buffer = receive_msg(socket);
                       ^ ~~~~~~~~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:386:2: warning: implicit declaration of function 'send_msg' is invalid in C99 [-Wimplicit-function-declaration]
        send_msg(&msg, socket); /* Send answer to erlang */
        ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:400:14: warning: incompatible pointer to integer conversion passing 'int (int, int, int)' to parameter of type 'int' [-Wint-conversion]
    shutdown(socket, 2);
             ^~~~~~
/usr/include/sys/socket.h:700:17: note: passing argument to parameter here
int     shutdown(int, int);
                    ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:1510:15: warning: unused variable 'result' [-Wunused-variable]
    SQLRETURN result;
              ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:1918:1: error: expected identifier or '('
{
^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2721:14: warning: incompatible pointer types passing 'SQLLEN **' (aka 'long **') to parameter of type 'SQLLEN *' (aka 'long *'); remove & [-Wincompatible-pointer-types]
                        blocklen, &StrLen_or_IndPtr);
                                  ^~~~~~~~~~~~~~~~~
/usr/local/include/sql.h:706:43: note: passing argument to parameter 'StrLen_or_Ind' here
                                  SQLLEN *StrLen_or_Ind);
                                          ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2724:41: warning: comparison between pointer and integer ('SQLLEN *' (aka 'long *') and 'int')
    fetched_bytes = (((StrLen_or_IndPtr == SQL_NO_TOTAL) || (StrLen_or_IndPtr > blocklen))
                       ~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2724:79: warning: ordered comparison between pointer and integer ('SQLLEN *' (aka 'long *') and 'int')
    fetched_bytes = (((StrLen_or_IndPtr == SQL_NO_TOTAL) || (StrLen_or_IndPtr > blocklen))
                                                             ~~~~~~~~~~~~~~~~ ^ ~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2725:13: warning: pointer/integer type mismatch in conditional expression ('int' and 'SQLLEN *' (aka 'long *')) [-Wconditional-type-mismatch]
            ? (blocklen-1) : StrLen_or_IndPtr);
            ^ ~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2724:19: warning: incompatible pointer to integer conversion assigning to 'int' from 'SQLLEN *' (aka 'long *') [-Wint-conversion]
    fetched_bytes = (((StrLen_or_IndPtr == SQL_NO_TOTAL) || (StrLen_or_IndPtr > blocklen))
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2743:5: warning: incompatible pointer types passing 'SQLLEN **' (aka 'long **') to parameter of type 'SQLLEN *' (aka 'long *'); remove & [-Wincompatible-pointer-types]
                                &StrLen_or_IndPtr);
                                ^~~~~~~~~~~~~~~~~
/usr/local/include/sql.h:706:43: note: passing argument to parameter 'StrLen_or_Ind' here
                                  SQLLEN *StrLen_or_Ind);
                                          ^
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2745:45: warning: comparison between pointer and integer ('SQLLEN *' (aka 'long *') and 'int')
        fetched_bytes = (((StrLen_or_IndPtr == SQL_NO_TOTAL) || (StrLen_or_IndPtr > blocklen))
                           ~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2745:83: warning: ordered comparison between pointer and integer ('SQLLEN *' (aka 'long *') and 'int')
        fetched_bytes = (((StrLen_or_IndPtr == SQL_NO_TOTAL) || (StrLen_or_IndPtr > blocklen))
                                                                 ~~~~~~~~~~~~~~~~ ^ ~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2746:17: warning: pointer/integer type mismatch in conditional expression ('int' and 'SQLLEN *' (aka 'long *')) [-Wconditional-type-mismatch]
                ? (blocklen-1) : StrLen_or_IndPtr);
                ^ ~~~~~~~~~~~~   ~~~~~~~~~~~~~~~~
/Users/michalpiotrowski/projects/MongooseIM/repo/_build/default/lib/eodbc/c_src/odbcserver.c:2745:23: warning: incompatible pointer to integer conversion assigning to 'int' from 'SQLLEN *' (aka 'long *') [-Wint-conversion]
        fetched_bytes = (((StrLen_or_IndPtr == SQL_NO_TOTAL) || (StrLen_or_IndPtr > blocklen))
                      ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
18 warnings and 3 errors generated.```

Reply:
Also noticed by @kzemek. We're going to assign a ticket for this in upcoming sprint.